doc/sphinx/addendum/universe-polymorphism.rst

```diff
@@ -270,30 +270,74 @@ Insufficiently restrictive variance annotations lead to errors:
 
    Fail Polymorphic Cumulative Record bad@{*a} := {p : Type@{a}}.
 
-An example of a proof using cumulativity
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-.. coqtop:: in reset
-
-   Set Universe Polymorphism.
-   Set Polymorphic Inductive Cumulativity.
-
-   Inductive eq@{i} {A : Type@{i}} (x : A) : A -> Type@{i} := eq_refl : eq x x.
-
-   Definition funext_type@{a b e} (A : Type@{a}) (B : A -> Type@{b})
-   := forall f g : (forall a, B a),
-                   (forall x, eq@{e} (f x) (g x))
-                   -> eq@{e} f g.
-
-   Section down.
-      Universes a b e e'.
-      Constraint e' < e.
-      Lemma funext_down {A B}
-        (H : @funext_type@{a b e} A B) : @funext_type@{a b e'} A B.
-      Proof.
-        exact H.
-      Defined.
-   End down.
+.. example:: Demonstration of universe variances
+
+   .. coqtop:: in
+
+      Set Printing Universes.
+      Set Universe Polymorphism.
+      Set Polymorphic Inductive Cumulativity.
+
+      Inductive Invariant @{=u} : Type@{u}.
+      Inductive Covariant @{+u} : Type@{u}.
+      Inductive Irrelevent@{*u} : Type@{u}.
+
+      Section Universes.
+        Universe low high.
+        Constraint low < high.
+
+        (* An invariant universe blocks cumulativity from upper or lower levels. *)
+        Axiom inv_low  : Invariant@{low}.
+        Axiom inv_high : Invariant@{high}.
+   .. coqtop:: all
+
+        Fail Check (inv_low : Invariant@{high}).
+        Fail Check (inv_high : Invariant@{low}).
+   .. coqtop:: in
+
+        (* A covariant universe allows cumulativity from a lower level. *)
+        Axiom co_low  : Covariant@{low}.
+        Axiom co_high : Covariant@{high}.
+   .. coqtop:: all
+
+        Check (co_low : Covariant@{high}).
```

Comment:
The last line of output for this `Check` seems odd.  I don't see anything similar for the other cases.  WDYT?  Not something to change in this PR, just an observation.

![image](https://user-images.githubusercontent.com/1253341/155200192-044bfc82-8cf4-41a0-8a37-758a935091e1.png)

Reply:
That's from `Set Printing Universes`, since the output has at least two different universes, their constrains are also printed at the bottom. (I think).

Reply:
With Set Printing Universes, Check prints the constraints which were generated while checking the term. When combined with preexisting constraints they may be redundant or imply additional constraints by transitivity.